app/services/idv/financials_step.rb

```diff
@@ -1,27 +1,30 @@
 module Idv
   class FinancialsStep < Step
-    def complete?
-      idv_form.finance_type && idv_session.financials_confirmation.try(:success?) ? true : false
+    def call
+      if complete?
+        @success = true
+        idv_session.financials_confirmation = true
+        idv_session.params = idv_form.idv_params
```

Comment:
This makes it easier to see what happens in the success versus failed scenario, as opposed to hiding the side effects in various private methods.